### PR TITLE
Have the bot chatter repeat sig mentions when non org member creates

### PR DESCRIPTION
**Pull request by grodrigues3**
Non org members can't mention sigs right now and their issues are often less unrouted.  This will have the bot (which is an org member) reiterate their mention and get a sigs attention.

@davidopp @kubernetes/sig-contributor-experience-feature-requests

### CODE REVIEW

**Comment by davidopp** [comment]
I can't comment on the correctness of this since I don't know the prow code, but thanks for doing this!

Just so I understand, this is not actually adding a new slash command, but will scan for any @ mentions in the body and repeat them?

**Comment by grodrigues3** [comment]
Yeah, not a new slash command. @spiffxp is working on that (so we have symmetry and a way to remove sig labels to).

This is just about repeating non K8s users attempts at @mentioning.  

TODO:  Add an issue that encourages people to @ mention a sig on issue creation.

File: prow/plugins/label/label.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ var (
 	labelRegex              = regexp.MustCompile(`(?m)^/(area|priority|kind)\s*(.*)$`)
 	removeLabelRegex        = regexp.MustCompile(`(?m)^/remove-(area|priority|kind)\s*(.*)$`)
 	sigMatcher              = regexp.MustCompile(`(?m)@kubernetes/sig-([\w-]*)-(?:misc|test-failures|bugs|feature-requests|proposals|pr-reviews|api-reviews)`)
```
Comment (grodrigues3) [inline prow/plugins/label/label.go]: @spxtr Out of curiosity, do you now why it's called `assignEvent`?

File: prow/plugins/label/label.go [hunk 2]
```diff
@@ -194,9 +197,19 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent) error {
 		}
 		if err := gc.AddLabel(ae.org, ae.repo, ae.number, sigLabel); err != nil {
 			log.WithError(err).Errorf("Github failed to add the following label: %s", sigLabel)
+		} else if !isMember {
+			toRepeat = append(toRepeat, sigMatch[0])
 		}
 	}
 
+	if len(toRepeat) > 0 {
+		msg := fmt.Sprintf(chatBack, strings.Join(toRepeat, ", "))
```
Comment (crimsonfaith91) [inline prow/plugins/label/label.go]: Correct me if i am wrong.
This message will run like: 
`Reiterating the mentions to trigger a notification: \n @kubernetes/sig-bugs, @kubernetes/sig-misc`

File: prow/plugins/label/label.go [hunk 3]
```diff
@@ -194,9 +197,19 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent) error {
 		}
 		if err := gc.AddLabel(ae.org, ae.repo, ae.number, sigLabel); err != nil {
 			log.WithError(err).Errorf("Github failed to add the following label: %s", sigLabel)
+		} else if !isMember {
+			toRepeat = append(toRepeat, sigMatch[0])
```
Comment (crimsonfaith91) [inline prow/plugins/label/label.go]: a unit test for verifying `toRepeat` items will be great :)

**Comment by crimsonfaith91** [comment]
the codes look good to me. :)

/lgtm

File: prow/plugins/label/label.go [hunk 4]
```diff
@@ -117,6 +118,21 @@ func getLabelsFromREMatches(matches [][]string) (labels []string) {
 	return
 }
 
+func (ae assignEvent) getRepeats(sigMatches [][]string, existingLabels map[string]string) (toRepeat []string) {
```
Comment (grodrigues3) [inline prow/plugins/label/label.go]: This repeats a lot of the functionality from the for loop below that actually adds the labels.  That said, it order to make it modular and testable, this is what I came up with

Comment (crimsonfaith91) [inline prow/plugins/label/label.go]: great to have it modular! :)

**Comment by grodrigues3** [comment]
I'll just wait for @spxtr to comment on whether I need to bump or update anything.

File: prow/plugins/label/label_test.go [hunk 5]
```diff
@@ -208,6 +197,16 @@ func getFakeRepoPullRequest(commentBody, commenter string, repoLabels, issueLabe
 func TestLabel(t *testing.T) {
 	// "a" is the author, "a", "r1", and "r2" are reviewers.
 
+	type testCase struct {
```
Comment (crimsonfaith91) [inline prow/plugins/label/label_test.go]: Apart from easier reference, is there a reason declaring struct in a function is better than doing so at beginning of the file?

File: prow/plugins/label/label_test.go [hunk 6]
```diff
@@ -553,3 +552,121 @@ func TestLabel(t *testing.T) {
...
+		{
+			name: "Repeat multiple valid labels from non org member",
+			body: "@kubernetes/sig-node-misc @kubernetes/sig-api-machinery-bugs",
+			expectedRepeatedLabels: []string{"@kubernetes/sig-node-misc", "@kubernetes/sig-api-machinery-bugs"},
+			repoLabels:             []string{"sig/node", "sig/api-machinery"},
+			issueLabels:            []string{},
+			commenter:              nonOrgMember,
+		},
+		{
+			name: "Repeat multiple valid labels from non org member",
```
Comment (crimsonfaith91) [inline prow/plugins/label/label_test.go]: Perhaps a better name for the test case is `Repeat multiple valid labels with a line break from non org member`.

File: prow/plugins/label/label_test.go (hunk 5 above)
Comment (grodrigues3) [inline prow/plugins/label/label_test.go]: not every test will require the same testCase struct type.  For example, in TestRepeat, we don't need `expectedNewLabels, expectedRemovedLabels`.

File: prow/plugins/label/label.go [hunk 7]
```diff
@@ -126,6 +142,7 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent) error {
 	labelMatches := labelRegex.FindAllStringSubmatch(ae.body, -1)
 	removeLabelMatches := removeLabelRegex.FindAllStringSubmatch(ae.body, -1)
 	sigMatches := sigMatcher.FindAllStringSubmatch(ae.body, -1)
+	isMember, _ := gc.IsMember(ae.org, ae.login)
```
Comment (rmmh) [inline prow/plugins/label/label.go]: can you only make this call if there are sigMatches? *most* comments won't mention a sig at all, so wasting an API call checking for membership is pointless.

File: prow/plugins/label/label.go (hunk 2 above)
Comment (spiffxp) [inline prow/plugins/label/label.go]: I'm not sure those teams exist.  The idea is to match @kubernetes/sig-foo-api-reviews, add the sig/foo label, and re-comment @kubernetes/sig-foo-api-reviews

Comment (spiffxp) [inline prow/plugins/label/label.go]: whoops, think you got it correct now, I must've been looking at an outdated change

File: prow/plugins/label/label_test.go (hunk 5 above)
Comment (fejta) [inline prow/plugins/label/label_test.go]: If something is only used in a local function it is better to keep the definition within that local function so someone looking outside that function doesn't have to spend the cognitive resources determining whether this is something they need to understand.

Comment (crimsonfaith91) [inline prow/plugins/label/label_test.go]: @fejta @grodrigues3 noted! thank you very much for the feedback! :)

File: prow/plugins/label/label.go [hunk 8]
```diff
@@ -183,6 +199,11 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent) error {
 		}
 	}
 
+	if len(sigMatches) == 0 {
```
Comment (rmmh) [inline prow/plugins/label/label.go]: Maybe guard the call below directly?

    toRepeat := []string{}
    if len(sigMatches) > 0 {
        isMember, _ := gc.IsMember(ae.org, ae.login)
        if !isMember {
            toRepeat = ae.getRepeats(sigMatches, existingLabels)
        }
    }

**Comment by grodrigues3** [comment]
squashing and merging.

**Pull request merged**